bash-preexec.sh

```diff
@@ -342,8 +371,33 @@ __bp_install_after_session_init() {
     if [[ -n "$sanitized_prompt_command" ]]; then
         PROMPT_COMMAND=${sanitized_prompt_command}$'\n'
     fi;
-    PROMPT_COMMAND+=${__bp_install_string}
+    PROMPT_COMMAND+=${bash_preexec_install_string}
+}
+
+# Remove hooks installed in the DEBUG trap and PROMPT_COMMAND.
+bash_preexec_uninstall() {
```

Comment:
I'm not convinced that bash-preexec should be responsible for maintaining any over head of uninstalling itself from both trap and prompt_command. Any libraries using these shared variables have more or less had to clean them up and maintain them on their own, as does bash-preexec with its installation. With the defined global variables what's stopping from just implementing this in your own library?

Reply:
> With the defined global variables what's stopping from just implementing this in your own library?

If the setup of `bash-preexec.sh` would never change in the future forever, I could just assume the current setup of `bash-preexec.sh` and write a code to clean up the things that are installed by the current `bash-preexec.sh`.

However, if the `__bp_install` configuration can change in the future (e.g. as suggested in #128), the cleanup code also needs to be updated. I don't think it is a good idea to update the cleanup code at the users' side every time the `__bp_install` configuration of `bash-preexec.sh` is changed. In particular, if the method of uninstalling the hooks would not be provided by `bash-preexec.sh`, the user/library needs to maintain the separate cleanup code for each of *all* the past versions of `bash-preexec.sh` and to detect the precise version of `bash-preexec.sh`.

This is actually the reason that I would like to request the "stable" public API rather than relying on the details of the internal implementation that can be frequently changed.

Reply:
I get your reasoning and desire for this as a way to help with your library and not just implementing it yourself. I just think there's no real standard here I've seen other libraries that have clashed with bash-preexec as well since they also make the assumption they'll have access to these parts of the system configuration. ([Liquid Prompt](https://github.com/nojhan/liquidprompt) is one example) My assumption is, these too can cause issues with ble.sh and we're trying to chase individual fires here.  

I am concerned that this is an esoteric use case and will likely just add operational overhead to maintaining it. With that said, if any other contributors see value in this I'm supportive.

Reply:
Thanks. Maybe I miss your exact point (What does "the system configuration" mean?), but I think I'll later need to look at Liquid Prompt.